Sources/SwifterSwift/CoreLocation/CLLocationArrayExtensions.swift

```diff
@@ -0,0 +1,30 @@
+//
+//  CLLocationArrayExtensions.swift
+//  SwifterSwift
+//
+//  Created by Trevor Phillips on 09/01/20.
+//  Copyright © 2020 SwifterSwift
+//
+
+#if canImport(CoreLocation)
+import CoreLocation
+
+// MARK: - Methods
+public extension Array where Element: CLLocation {
+
+    /// SwifterSwift: Calculates the sum of distances between each location in the array based on the curvature of the earth.
+    ///
+    /// - Parameter unit: The unit of length to return the distance in.
+    /// - Returns: The distance in the specified unit.
+    @available(iOS 10.0, tvOS 10.0, macOS 10.12, watchOS 3.0, *)
+    func distance(unit: UnitLength) -> Measurement<UnitLength> {
+        let distance = self.enumerated().reduce(0.0) {
+            if $1.1 == self.last { return $0 }
```

Comment:
To make this code more let's use named variables here, and if possible instead of tuple indexes use labels :))

Reply:
@LucianoPAlmeida sorry could you give an example, not quite sure what you mean?

named variables such as: `distance(unitLength unit: UnitLength)`?

and for tuple indexes you mean to use: `if $1.element == self.last { return $0 }` instead?

Reply:
In part yeah, but also replace the $0 an $1 to named variables :)) 
Also, one question if we have an empty or a one-element array? Can you add tests for those cases?

Reply:
@LucianoPAlmeida I think I did what you were asking for.

Reply:
That's nice @trevorphillips, thank you!
But taking a look in more detail even though I thought `reduce` was a good option here, it seems to make the code more complicated overall and we always have to check for last in the loop which maybe is not the better solution. 
What do you think about 
```swift
func distance(unitLength unit: UnitLength) -> Measurement<UnitLength> {
        guard count > 1 else {
          return Measurement(value: 0.0, unit: unit)
        }
        var distance = 0.0
        for idx in 0..<count-1 {
            distance += self[idx].distance(from: self[idx + 1])
        }
        return Measurement(value: distance, unit: UnitLength.meters).converted(to: unit)
}
```

Reply:
@LucianoPAlmeida yeah works for me. just updated my code.

Reply:
Nice, Thank you :)